retrofit/src/test/java/retrofit2/RequestBuilderTest.java

```diff
@@ -2578,6 +2579,44 @@
     assertThat(readBody.indexOf("secondParam")).isLessThan(readBody.indexOf("thirdParam"));
   }
 
+
+  @Test public void queryParamsSkippedIfConvertedToNull() throws IOException, InterruptedException {
+    class Example {
+      @POST("/query") Call<String> queryPath(@Query("a") Object a) { return null; }
+    }
+
+    Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
+            .baseUrl("http://example.com")
+            .addConverterFactory(new NullObjectConverterFactory());
+
+    Request request = buildRequest(Example.class, retrofitBuilder, "Ignored");
+
+    if(request.url().toString().contains("Ignored")){
+      fail();
+    }
+  }
+
+  @Test public void queryParamMapsConvertedToNullShouldError() throws IOException, InterruptedException {
+    class Example {
+      @POST("/query") Call<String> queryPath(@QueryMap HashMap<String, String> a) { return null; }
+    }
+
+    Retrofit.Builder retrofitBuilder = new Retrofit.Builder()
+            .baseUrl("http://example.com")
+            .addConverterFactory(new NullObjectConverterFactory());
+
+    final HashMap<String, String> queryMap = new HashMap<String, String>() {{
```

Comment:
just btw, [always prefer LinkedHashMap](https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/).